### PR TITLE
Akismet spam filter

### LINKED ISSUE #4: Foiling the bots with Akismet
1) When the server starts it will create a object with a set of akismet clients as it's properties.

```js
const akismet = require('akismet-api')
const config = require('config')

const tableflip = akismet.client({
  key  : '{config.akismet.tableflip.key}',
  blog : 'http://tableflip.io'
})

const anotherSite = akismet.client({
  key  : '{config.akismet['another-site'].key}',
  blog : 'http://another-site.io'
})

module.exports = {
  "tableflip.io": tableflip,
  "another-site.co.uk": anotherSite
}

```

In-coming posts will be caught in [findRoute](https://github.com/tableflip/post/blob/master/pages/home/server.js#L11) and we'll call out to the Akismet API like this

```js
const akismet = require('post-askimet-clients')

akismet[domain].checkSpam({
  user_ip: req.connection.remoteAddress,
  user_agent: req.headers['user-agent'],
  referrer: domain,
})
.then((spam) => if (spam) return //otherwise carry on)
.catch((err) => if (err) throw your rattle)

### PR DESCRIPTION
**Pull request by bmordan**
Resolves #4 

This PR adds a spam checker.

1) Posts come in from the client and are saved to the db ➡️
2) Both `mailer` and `spam` are listening to `put` events on the db so both are triggered ➡️
3) If the `value` being written is missing an `X-Spam-Flag` the `mailer` event returns,  while the `spam` event makes a call to Akismet ➡️
4) The result of calling out to Akismet is stored on an `X-Spam-Flag` property on the `value` and this update is written to the db triggering a `put` event ➡️
5) Both `mailer` and `spam` are listening to `put` events so both are triggered ➡️
6) Now `value` has an `X-Spam-Flag` so the `spam` function returns and `mailer` checks if the `X-Spam-Flag` property ➡️
7) If 'YES' the mailer removes the message ⏹
8) If 'NO' the message is sent ⏹

### CODE REVIEW

**Comment by bmordan** [comment]
dependency-ci fails because of jade

File: emails/spam.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    var client = akismet.client({
+      key: config.akismet.apiKey,
+      blog: 'http://' + value.key
+    })
+
+    client.checkSpam({
+      user_ip: value.headers['user-ip'],
+      user_agent: value.headers['user-agent'],
+      referrer: value.key
```
Comment (olizilla) [inline emails/spam.js]: This should probably be the referrer header from the request

File: emails/mailer.js [hunk 2]
```diff
@@ -12,11 +12,20 @@ module.exports = function init (db) {
     if (key.indexOf('msg!') !== 0) return
     if (!value.route) return
     if (!value.route.email) return
+    if (!value.headers['X-Spam-Flag']) return
```
Comment (olizilla) [inline emails/mailer.js]: Check for property existance
```js
value.headers.hasOwnProperty('X-Spam-Flag')
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty

**Review by olizilla — CHANGES_REQUESTED** [review]
Minor changes. Is good stuff.

**Pull request merged**